src/System.IO.Compression.Brotli/System/IO/Compression/Brotli/BrotliPrimitives.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Runtime.CompilerServices;
+using System.Text;
+
+#if BIT64
+    using nuint = System.UInt64;
+#else
+    using nuint = System.UInt32;
+#endif 
+
+namespace System.IO.Compression
+{
+    public struct BrotliPrimitives
+    {
+        const int defQuality = 11;
+        const int defLgWin = 24;
+
+        public static bool Compress(ReadOnlySpan<byte> source, Span<byte> destination, out nuint bytesConsumed, out nuint bytesWritten)
```

Comment:
@Vedin that's not a great argument in .NET. While we all agree that size cannot be less than zero, not all .NET languages support unsigned integers. Besides, I believe @ahsonkhan is asking if the size needs to be `size_t` or if the signature can just be fixed.